src/lib/autoinstall/dialogs/storage.rb

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cwm"
+require "cwm/dialog"
+require "autoinstall/widgets/storage/overview_tree_pager"
+require "autoinstall/storage_controller"
+
+Yast.import "Label"
+Yast.import "Popup"
+
+module Y2Autoinstallation
+  module Dialogs
+    # Dialog to set up the partition plan
+    #
+    # @example Edit a partitioning section
+    #   devicegraph = devicegraph = Y2Storage::StorageManager.instance.probed
+    #   partitioning = Y2Storage::Autoinst::PartitioningSection.new_from_storage(devicegraph)
+    #   result = Y2Autoinstallation::Dialogs::Storage.new(partitioning).run
+    class Storage < CWM::Dialog
+      # @return [Y2Storage::AutoinstProfile::PartitioningSection]
+      #   Partitioning section of the profile
+      attr_reader :partitioning
+
+      # Constructor
+      #
+      # @param partitioning [Y2Storage::AutoinstProfile::PartitioningSection]
+      #   Partitioning section of the profile
```

Comment:
maybe here it should mention that it set initial state of partitioning. Can I somehow pass empty partitioning?

Reply:
Yes, you should be able to pass an empty partitioning section (`Y2Storage::AutoinstProfile::PartitioningSection.new`). To make it clear, I have set a default value for `partitioning`.

Currently, there is a bug that prevents the dialog to work with an empty section, but it will be fixed soon.